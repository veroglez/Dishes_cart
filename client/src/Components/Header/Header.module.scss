.header {
  border-bottom: 1px solid #c8c8c8;
  height: 64px; 
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 0 20px;
  position: fixed;
  width: calc(100% - 40px);
  background: white;
  top: 0;

  .cart {
    width: 50px;
    margin-right: 12px;
    cursor: pointer;
  }

  .number {
    font-size: 24px;
    margin: 0;
  }
}