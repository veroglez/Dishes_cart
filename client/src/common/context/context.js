import { node } from 'prop-types';
import React, { createContext, Component } from 'react';
import l10n from '../l10n';

const Context = createContext();
const { Provider, Consumer: ConsumerData } = Context;

class ProviderData extends Component {
  static propTypes = {
    children: node,
  };

  static defaultProps = {
    children: undefined,
  };

  state = {
    l10n,
    products: [],
    items: 0,
    total: 0,
  }

  onData = object => this.setState(object);

  render() {
    const {
      onData, props: { children },
    } = this;

    return (
      <Provider value={{ ...this.state, onData }}>
        { children }
      </Provider>
    );
  }
}

export { ConsumerData, ProviderData };