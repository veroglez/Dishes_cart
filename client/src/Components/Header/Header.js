import React, { PureComponent } from 'react';
import { ConsumerData } from '../../common/context/context';
import { Modal } from '..';
import assets from '../../common/assets';
import style from './Header.module.scss';

class Card extends PureComponent {
  state = { modal: false };

  onModal = (value) => {
    document.body.style.overflow = value ? 'hidden' : 'initial';
    this.setState({ modal: value });
  }

  render() {
    const {
      onModal, state: { modal },
    } = this;

    return (
      <ConsumerData>
        {({ products }) => (
          <div className={style.header}>
            <img
              alt=""
              className={style.cart}
              onClick={() => onModal(true)}
              src={assets.cart}
            />
            <p className={style.number}>{products.length}</p>
            {modal && (
              <Modal onClose={onModal} />
            )}
          </div>
        )}
      </ConsumerData>
    );
  }
}

export default Card;
