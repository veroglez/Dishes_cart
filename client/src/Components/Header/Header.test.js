import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import { ProviderData } from '../../common/context/context';

const PRODUCTS = [
  { name: 'lorem ipsum 1', image: '/image.png', price: 20 },
  { name: 'lorem ipsum 2', image: '/image.png', price: 30 },
];

const HOC = inherit => (
  <ProviderData>
    <Header {...PRODUCTS} {...inherit} />
  </ProviderData>
);

describe('<Header>', () => {
  it('when {dishes}', () => {
    const tree = renderer.create(<HOC />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
