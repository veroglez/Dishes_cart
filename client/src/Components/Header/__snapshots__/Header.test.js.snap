// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> when {dishes} 1`] = `
<div
  className="header"
>
  <img
    alt=""
    className="cart"
    onClick={[Function]}
    src=""
  />
  <p
    className="number"
  >
    0
  </p>
</div>
`;
